iris_grib/tests/unit/load_convert/test_data_representation_section.py

```diff
@@ -0,0 +1,38 @@
+# Copyright iris-grib contributors
+#
+# This file is part of iris-grib and is released under the LGPL license.
+# See COPYING and COPYING.LESSER in the root of the repository for full
+# licensing details.
+"""
+Test function :func:`iris_grib._load_convert.data_representation_section.`
+
+"""
+
+# import iris_grib.tests first so that some things can be initialised
+# before importing anything else.
+import iris_grib.tests as tests
+
+from iris.exceptions import TranslationError
+
+from iris_grib._load_convert import data_representation_section
+from iris_grib.tests.unit import _make_test_message
+
+
+class Test(tests.IrisGribTest):
+    def test_supported_templates(self):
+        template_nums = [0, 1, 2, 3, 4, 40, 41, 42, 50, 51, 61]
+        for template_num in template_nums:
+            message = _make_test_message(
+                {5: {'dataRepresentationTemplateNumber': template_num}})
+            data_representation_section(message.sections[5])
```

Comment:
I admit this isn't very unit-ty but I prioritised maintainability, and ease-of-reading over unit-tiness. 

Alternatives I considered were:
* following [what we do for testing time range indicator](https://github.com/SciTools/iris-grib/blob/master/iris_grib/tests/unit/grib1_load_rules/test_grib1_convert.py#L61-L101) but that seemed to be too much boiler plate
* dynamically adding tests to the test class `Test` with `setattr` but I thought that may get complicated and harder to read??

Reply:
I don't know the specifics for iris-grib, but my general opinion on this subject is:
* Developers spend more time maintaining code than writing it so [**readability counts**](https://zen-of-python.info/readability-counts.html).
* While less unit-y tests don't provide that instant, highly specific information on a failure, it usually doesn't take long to manually work that out.

So for me, favouring test readability over result specificity is usually a good trade-off.

Reply:
I really don't have a problem with this.